### ISSUE TITLE
`extern` commands should still be treated as externals

### ISSUE THREAD

**Issue by amtoine** [nushell/nushell#9699]
### Describe the bug

with the @nushell/core-team, we had this idea of highlighting the external commands with a clearly distinct theme to avoid confusion between internals and externals.

this works fine apart from `extern` commands, mainly from completion file, which are treated as internals by Nushell.

### How to reproduce

in the following we will *run* the following pipeline which mixes internal (`each`, `lines` and `where`) with external (`git branch` and `git bar`) commands
```nushell
git branch -a | each {|it| $it | lines | where some == condition; git bar --another-option }
```

> **Note**  
> this pipeline does not make real sense, the `git bar` does not exist for instance, but it demonstrates the point about highlighting

1. run without any particular config, e.g. in `nu -n`

![external_hightlight no-config](https://github.com/nushell/nushell/assets/44101798/914ed406-983c-4765-8308-002cc4ce9fa1)

2. define the theme with hightlighting
```nushell
$env.config.color_config = {
    shape_external: light_green_reverse
    shape_externalarg: light_purple_reverse
    shape_internalcall: cyan_bold
}
```
![external_hightlight config](https://github.com/nushell/nushell/assets/44101798/9c2aeaba-344e-4b9e-b2c1-5e3952f128a1)

3. define `git branch` as a know external with
```nushell
extern "git branch" []
```
![external_hightlight extern](https://github.com/nushell/nushell/assets/44101798/9ed596ec-9223-43c2-908f-90a77581dbd7)

### Expected behavior

i expected the last one to render `git branch` as an external command, even though it's now *known* with `extern`, it's still an external command :yum: 
![external_hightlight config](https://github.com/nushell/nushell/assets/44101798/8221e569-3372-4f3c-86e1-3ae501c756be)

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.82.1                                   |
| branch             | main                                     |
| commit_hash        | 57d96c09fa78f6231093cb241da4ddbd545f84bf |
| build_os           | linux-x86_64                             |
| build_target       | x86_64-unknown-linux-gnu                 |
| rust_version       | rustc 1.69.0 (84c898d65 2023-04-16)      |
| rust_channel       | 1.69.0-x86_64-unknown-linux-gnu          |
| cargo_version      | cargo 1.69.0 (6e9a83356 2023-04-12)      |
| build_time         | 2023-07-15 12:59:41 +02:00               |
| build_rust_channel | release                                  |
| allocator          | standard                                 |
| features           | default, sqlite, trash, which, zip       |
| installed_plugins  | hist                                     |

**Comment by kubouch** [comment]
I think being able to discern between known and unknown externals is useful. If you want them to look the same, maybe you can update the color config to show both in the same style?

**Comment by amtoine** [comment]
> I think being able to discern between known and unknown externals is useful.
>
:+1: 

> If you want them to look the same, maybe you can update the color config to show both in the same style?
>
i do not see different styles in the color_config fields :confused: 

afaik, know externals are seen as internal commands by Nushell, thus they have the same colors, which is in turn different from the ones for externals :thinking:

**Comment by kubouch** [comment]
Ah, I see, sorry. Then, maybe being able to differentiate between those could be considered. Same for aliases, which are also internal commands from Nushell's point of view. This could probably be done by extending `FlatShape` to contain known external and alias variants. Then, during flattening, add the variants [here](https://github.com/nushell/nushell/blob/5bfec2024425f4a0cfb8400ed3d25b83cd6d1fe2/crates/nu-parser/src/flatten.rs#L190) based on `.is_alias()` or `.is_known_external()` result. Should be relatively straightforward.

I personally prefer having the known externals highlighted as internal because it lets me know that I have the support for custom completions etc.

**Comment by amtoine** [comment]
> Ah, I see, sorry. Then, maybe being able to differentiate between those could be considered. Same for aliases, which are also internal commands from Nushell's point of view. This could probably be done by extending `FlatShape` to contain known external and alias variants. Then, during flattening, add the variants [here](https://github.com/nushell/nushell/blob/5bfec2024425f4a0cfb8400ed3d25b83cd6d1fe2/crates/nu-parser/src/flatten.rs#L190) based on `.is_alias()` or `.is_known_external()` result. Should be relatively straightforward.
> 
yeah that'd be great because i have the same issue about aliases being seen as internal commands in https://github.com/nushell/nushell/pull/9646 :thinking: 

> I personally prefer having the known externals highlighted as internal because it lets me know that I have the support for custom completions etc.
>
but they are not internals :confused:

**Comment by crides** [comment]
given that we have resolved externals highlighting now, `extern`ed commands being highlighted the way they are basically blocks the resolvedness of the commands. Same issue (but more important I feel) with obscuring `which`

**Comment by fdncred** [comment]
It would be great to color aliases and custom commands as something different than internal calls.
